twitch.cpp

```diff
@@ -67,9 +71,24 @@ std::set<std::string> Twitch::getOnlineUsers(const std::string& channel) {
         }
     }
 
+    auto current_time = std::chrono::system_clock::now();
+    auto user_snapshot = _recent_chatters;
+    _recent_chatters = {};
+    for(auto const& [user, last_heard_from] : user_snapshot) {
+        if (current_time - last_heard_from < std::chrono::minutes(5)) {
+            online_users.emplace(user);
+            _recent_chatters.emplace(user, last_heard_from);
```

Comment:
Hm, it seems like it might be a little extra - do we really need to clean it up? Even a few hundred independent viewers per run isn't so bad in one map.

Reply:
Ah, you can replace these two lines with a call to markAsOnline() too.

Reply:
Good point, but this does use a different time than markAsOnline.  Also, the first line can't be replaced, only the second.